File: src/main/java/org/yearobjectives/domain/entity/ObjectiveType.java
package org.yearobjectives.domain.entity;

import java.time.Instant;

import org.yearobjectives.domain.entity.util.DateUtils;
import org.yearobjectives.domain.entity.util.MarkerTypeDates;

public enum ObjectiveType {

    DAILY(new DateUtils.Day()),
    WEEKLY(new DateUtils.Week()),
    MONTHLY(new DateUtils.Month()),
    YEARLY(new DateUtils.Year());
    
    private final MarkerTypeDates markerTypeDates;

    ObjectiveType(MarkerTypeDates markerTypeDates) {
        this.markerTypeDates = markerTypeDates;
    }

    public Instant getStart() {
        return this.markerTypeDates.getStart();
    }

    public Instant getEnd() {
        return this.markerTypeDates.getEnd();
    }
    
}

File: src/main/kotlin/com/objectives/management/domain/entity/ObjectiveType.kt
package com.objectives.management.domain.entity

import com.objectives.management.domain.entity.utils.DateTypeRange
import com.objectives.management.domain.entity.utils.DateUtils
import java.time.Instant

enum class ObjectiveType(private val dateTypeRange: DateTypeRange) {
    DAILY(DateUtils.Day()),
    WEEKLY(DateUtils.Week()),
    MONTHLY(DateUtils.Month()),
    YEARLY(DateUtils.Year());

    fun getStartDate(): Instant = dateTypeRange.getStartDate()

    fun getEndDate(): Instant = dateTypeRange.getEndDate()
}